client/ayon_harmony/api/plugin.py

```diff
@@ -32,32 +168,104 @@ def setup_node(self, node):
+        raise NotImplementedError
+
+    def get_pre_create_attr_defs(self):
+        output = [
+            BoolDef(
+                "use_selection",
+                tooltip="Composition for publishable instance should be "
+                        "selected by default.",
+                default=True,
+                label="Use selection"
+            ),
+        ]
+        return output
+
+
+class HarmonyRenderCreator(HarmonyCreator):
+
+    node_type = "COMPOSITE"
+    auto_connect = False
+    composition_node_pattern = ""
+
+    rendering_targets = {
+        "local": "Local machine rendering",
+        "farm": "Farm rendering",
+    }
+
+    def product_impl(self, name, instance_data: dict, pre_create_data: dict):
         existing_node_names = harmony.send(
             {
                 "function": "AyonHarmonyAPI.getNodesNamesByType",
-                "args": self.node_type
+                "args": [self.node_type]
             })["result"]
 
-        # Dont allow instances with the same name.
-        msg = "Instance with name \"{}\" already exists.".format(self.name)
-        for name in existing_node_names:
-            if self.name.lower() == name.lower():
-                harmony.send(
+        # Don't allow instances with the same name.
+        name_lower = name.lower()
+        for existing_name in existing_node_names:
+            if name_lower == existing_name.lower():
+                msg = f"Instance with name \"{name}\" already exists."
+                raise CreatorError(msg)
+
+        with harmony.maintained_selection() as selection:
+
+            args = [name, self.node_type]
+            if pre_create_data.get("use_selection") and selection:
+                args.append(selection[-1])
+            elif self.auto_connect:
+                existing_comp_names = harmony.send(
                     {
-                        "function": "AyonHarmonyAPI.message", "args": msg
-                    }
-                )
-                return False
+                        "function": "AyonHarmonyAPI.getNodesNamesByType",
+                        "args": "COMPOSITE"
+                    })["result"]
+                name_pattern = self.composition_node_pattern
+                if not name_pattern:
+                    raise CreatorError("Composition name regex pattern "
+                                       "must be filled")
+                compiled_pattern = re.compile(name_pattern)
+                matching_nodes = [name for name in existing_comp_names
+                                  if compiled_pattern.match(name)]
+                if len(matching_nodes) > 1:
+                    self.log.warning("Multiple composition node found, "
+                                     "picked first")
+                elif len(matching_nodes) <= 0:
+                    raise CreatorError("No matching composition "
+                                       "node found")
```

Comment:
If fact, to keep consistent behavior and be a bit more optimized, I'd move the selected node matching in the js container creation function. It'll reduce harmony API calls which are far too slow. Instead of passing a string you'd pass a bool for `useSelection` (like for `createBackdropContainer` and deal with the nodes there). 
NB: It'll also make `maintained_selection` obsolete and we can get rid of it.